src/hooks/usePDFPreview.ts

```diff
@@ -67,59 +66,58 @@ export default function usePDFPreview(physicalDocument: TDocumentPage, documentP
 
   const annotationConfig = { showToolbar: false, showCommentsPanel: false, downloadWithAnnotations: true, printWithAnnotations: true };
 
-  let viewSDKClient = null;
-  let embedApi = null;
-  let annotationManagerApi = null;
+  let viewSDKClient_CACHE = null;
+  let embedApi_CACHE = null;
 
-  const createPDFClient = (startingPassage: number, onLoadCallback?: Function) => {
-    viewSDKClient = new ViewSDKClient();
-    viewSDKClient.ready().then(() => {
-      // Prevent attaching preview to non-existent div
-      const pdfDiv = document.getElementById("pdf-div");
-      if (!pdfDiv) return;
-      const previewFilePromise = viewSDKClient.previewFile(physicalDocument, adobeKey, "pdf-div", viewerConfig);
-      previewFilePromise.then((adobeViewer: any) => {
-        // PDF PREVIEW -- SHOULD BE VISIBLE NOW
-        adobeViewer.getAPIs().then((api: any) => {
-          embedApi = api;
-          // if we have a passage index, scroll to it
-          if (!isNaN(Number(startingPassage))) {
-            passageIndexChangeHandler(startingPassage);
-          }
-        });
-        adobeViewer.getAnnotationManager().then((annotationManager: any) => {
-          annotationManager.setConfig(annotationConfig);
-          annotationManagerApi = annotationManager;
-          if (documentPassageMatches.length > 0) {
-            // annotationManager.addAnnotations(generateHighlights(document, documentPassageMatches));
-            documentMatchesChangeHandler(documentPassageMatches);
-          }
-        });
-        if (onLoadCallback) {
-          onLoadCallback();
-        }
-      });
-    });
+  const createPDFClient = async () => {
+    let viewSDKClient = null;
+    if (viewSDKClient_CACHE) {
+      viewSDKClient = viewSDKClient_CACHE;
+    } else {
+      viewSDKClient = new ViewSDKClient();
+      viewSDKClient_CACHE = viewSDKClient;
+    }
+
+    await viewSDKClient.ready();
+    const adobeViewer = await viewSDKClient.previewFile(physicalDocument, adobeKey, "pdf-div", viewerConfig);
+    const embedApi = await adobeViewer.getAPIs();
+    embedApi_CACHE = embedApi;
+    const annotationManagerApi = await adobeViewer.getAnnotationManager();
+    annotationManagerApi.setConfig(annotationConfig);
+    return {
+      embedApi,
+      annotationManagerApi,
+    };
   };
 
-  const passageIndexChangeHandler = (passageIndex: number) => {
+  const passageIndexChangeHandler = async (passageIndex: number, documentPassageMatches: TPassage[]) => {
+    let embedApi = embedApi_CACHE;
```

Comment:
thought: I think leaving the caching of the APIs to the above method would simplify this a bit by letting that method deal with the memo exclusively. Anything calling that method would just need to `await geAdobeApis`